### PR TITLE
Fixing composedSchema deserializing and resolving, allof, anyOf, Oneof

### PR DESCRIPTION
**Pull request by gracekarina**
The deserializing of composedSchema was only returning the last object of the arrays, and there were some issues in the resolving of this schemas.

### CODE REVIEW

File: modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java [hunk 1]
```diff
@@ -1785,20 +1785,18 @@ public Schema getSchema(ObjectNode node, String location, ParseResult result){
...
 
         if(allOfArray != null) {
-
+            ComposedSchema allOfList = new ComposedSchema();
             for(JsonNode n : allOfArray) {
                 if(n.isObject()) {
-                    ComposedSchema allOfList = new ComposedSchema();
                     schema = getSchema((ObjectNode) n,location,result);
                     allOfList.addAllOfItem(schema);
                     schema = allOfList;
```
Comment (phiz71) [inline modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java]: `schema = allOfList;` : this line should be outside the `for` structure

File: modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java [hunk 2]
```diff
@@ -1785,20 +1785,18 @@ public Schema getSchema(ObjectNode node, String location, ParseResult result){
...
             }
         } else if(anyOfArray != null) {
-
+            ComposedSchema anyOfList = new ComposedSchema();
             for(JsonNode n : anyOfArray) {
                 if(n.isObject()) {
-                    ComposedSchema anyOfList = new ComposedSchema();
                     schema = getSchema((ObjectNode) n,location,result);
                     anyOfList.addAnyOfItem(schema);
                     schema = anyOfList;
```
Comment (phiz71) [inline modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java]: `schema = anyOfList;` : this line should be outside the `for` structure

Comment (gracekarina) [reply modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java]: OMG It was.

File: modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java (hunk 1 above)
Comment (gracekarina) [reply modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java]: you are right

**Comment by phiz71** [comment]
I think the fixing of the `oneOf` case has been forgotten 😉

**Comment by gracekarina** [comment]
hehe done!

**Pull request merged**